### ISSUE TITLE
Web Extension: background script stops working in localhost when a content script is present

### ISSUE THREAD

**Issue by xkill-9** [parcel-bundler/parcel#9036]
# 🐛 bug report

When running a web extension locally through `parcel watch`, the `background` script stops working after a reload if there's a content script also defined in the `manifest.json`. If `content_scripts` is not set in the `manifest.json` the extension reloads and behaves as expected.

The bug only happens for `localhost` hosts, e.g: an app running in `http://localhost:3000`.

The bug does not happen when the extension is installed using the `dist` folder created by `parcel build`, all of this is reproducible following [parcel's web extension configuration docs](https://parceljs.org/recipes/web-extension/)

## 🎛 Configuration (.babelrc, package.json, cli command)

This is the minimal setup to reproduce the bug.

1. Create a folder for the extension, e.g: `my-extension` and add the following files to it:

- package.json
```json
{
  "name": "extension",
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "start": "parcel watch manifest.json --host localhost --config @parcel/config-webextension",
    "build": "parcel build manifest.json --config @parcel/config-webextension"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@parcel/config-webextension": "^2.8.3",
    "parcel": "^2.8.3"
  }
}
```
- manifest.json
```json
{
  "manifest_version": 3,
  "name": "Localhost Hello World",
  "version": "1.0",
  "description": "Testing Parcel autoload bug",
  "permissions": [
    "tabs",
    "scripting",
    "activeTab"
  ],
  "content_scripts": [
    {
      "matches": [
        "http://localhost/*"
      ],
      "run_at": "document_start",
      "js": ["content.js"]
    }
  ],
  "host_permissions": [
    "http://localhost/*"
  ],
  "background": {
    "service_worker": "background.js",
    "type": "module"
  }
}
```
- background.js
```js
const helloWorld = () => {
    console.log('Hello World')
}

chrome.tabs.onUpdated.addListener((id, change, tab) => {
    if (change.status === 'loading' && /localhost/.test(tab.url)) {
        chrome.scripting.executeScript({
            target: { tabId: id },
            func: helloWorld,
            args: [],
            injectImmediately: true,
            world: 'MAIN',
        })
    }
})
```
- content.js (the content of this file shouldn't make a difference)
```js
window.addEventListener('message', (e) => {
    console.log('message', e)
});
```

2. Navigate to the extension folder, install dependencies and run parcel's dev server with:
```
cd my-extension
yarn
yarn start
```

3. Install the extension by going to Chrome's Extensions Manager, clicking on "Load unpacked" and installing the `dist` folder that got created in the previous step.

4. To test the behavior, you also need to have an app running in localhost, I tested it with a brand new React app using vite:
```sh
yarn create vite app --template react
cd app
yarn
yarn dev
```

## 🤔 Expected Behavior

With the extension installed, **every** time you refresh the app the console should log:
```bash
Hello World
```

## 😯 Current Behavior

Refreshing the app, only logs `Hello World` in the console **once on the first refresh** after a full extension reload (through the Extensions Manager) any subsequent refresh will not log `Hello World`.

## 💁 Possible Solution

I don't have a solution but my guess is that the bug lies somewhere within the logic that injects the script that auto reloads the extension with every change when running the local dev server, or maybe the `autoreload` script itself is somehow breaking the `chrome.tabs.onUpdate.addListener` call. 

## 🔦 Context

I encountered this issue when working on a private dev tools extension that we use at work and noticing that some of the features weren't working as expected.

## 💻 Code Sample

All the steps to repro the bug are detailed in the Configuration section.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |   2.8.3
| Node             |  18.14.2
| npm/Yarn         |  9.5.0/1.22.19
| Operating System |  macOS Ventura 13.2

**Comment by Bhavyaatrivedi** [comment]
Hi, can I work on this issue.

**Comment by 101arrowz** [comment]
This is indeed consequence of autoreload - when you refresh a page that has a content script on it, the extension will fully reload itself. That means the `onUpdated` event never actually reaches the extension. This may be fixed by #9068 because there's now a delay before reloading the content script, but that will need testing.

**Comment by xkill-9** [comment]
@101arrowz I just checked, and this bug was indeed fixed by #9068, awesome work!
